webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala

```diff
@@ -386,7 +433,8 @@ class InternalLDAPQueryProcessor(
       }.map { case (ldapFilters, specialFilters) =>
         val finalLdapFilter = normalizedQuery.composition match {
           case Or  =>
-            Some(OR( (filterSeqSet.foldLeft(ldapFilters)( _ union _ )).toSeq:_*))
+            if( filterSeqSet.size+ldapFilters.size <= 0) Some(NOT(HAS(A_OC))) // means that we don't have any ldap filter, only extended one, so select no nodes, see #19538
```

Comment:
can it be negative? shouldn't it be
if( filterSeqSet.isEmpty && ldapFilters.isEmpty) Some(NOT(HAS(A_OC)))

Reply:
it could have been the `isEmpty` version. The perf of that, at that point, does matter (unlikelly it's an hot spot for anything)